ledger-core/virtual/integration/method_test.go

```diff
@@ -975,6 +977,201 @@ func TestVirtual_CallMethodFromConstructor_Unordered(t *testing.T) {
+	defer server.Stop()
+
+	executeDone := server.Journal.WaitStopOf(&execute.SMExecute{}, 1)
+
+	runnerMock := logicless.NewServiceMock(ctx, mc, func(execution execution.Context) string {
+		return execution.Request.CallSiteMethod
+	})
+	server.ReplaceRunner(runnerMock)
+	server.Init(ctx)
+	server.IncrementPulseAndWaitIdle(ctx)
+	var (
+		class = gen.UniqueGlobalRef()
+
+		objectAGlobal = reference.NewSelf(server.RandomLocalWithPulse())
+
+		flags = contract.MethodIsolation{
+			Interference: contract.CallIntolerable,
+			State:        contract.CallDirty,
+		}
+	)
+
+	Method_PrepareObject(ctx, server, payload.Ready, objectAGlobal)
+
+	outgoingCallRef := gen.UniqueGlobalRef()
+
+	expectedError, err := foundation.MarshalMethodErrorResult(throw.E("interference violation: ordered call from unordered call"))
+	require.NoError(t, err)
+
+	expectedResult := []byte("finish A.Foo")
+
+	outgoingCall := execution.NewRPCBuilder(outgoingCallRef, objectAGlobal).CallMethod(objectAGlobal, class, "Bar", []byte("123")).SetInterference(contract.CallTolerable)
+	objectAExecutionMock := runnerMock.AddExecutionMock("Foo")
+	objectAExecutionMock.AddStart(
+		func(ctx execution.Context) {
+			assert.Equal(t, objectAGlobal, ctx.Object)
+			assert.Equal(t, contract.MethodIsolation{
+				Interference: contract.CallIntolerable,
+				State:        contract.CallDirty,
+			}, ctx.Isolation)
+			t.Log("ExecutionStart [A.Foo]")
+		},
+		&execution.Update{
+			Type:     execution.OutgoingCall,
+			Error:    nil,
+			Outgoing: outgoingCall,
+		},
+	).AddContinue(func(result []byte) {
+		assert.Equal(t, expectedError, result)
+
+	}, &execution.Update{
+		Type:   execution.Done,
+		Result: requestresult.New(expectedResult, objectAGlobal),
+	})
+
+	runnerMock.AddExecutionClassify("Foo", flags, nil)
+
+	typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+	typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+		switch res.Callee {
+		case objectAGlobal:
+			assert.Equal(t, expectedResult, res.ReturnArguments)
+		default:
+			assert.Fail(t, "unexpected VCallResult")
+		}
+		return false
+	}).ExpectedCount(1)
+
+	pl := payload.VCallRequest{
+		CallType:            payload.CTMethod,
+		CallFlags:           payload.BuildCallFlags(contract.CallTolerable, contract.CallDirty),
+		Caller:              server.GlobalCaller(),
+		Callee:              objectAGlobal,
+		CallSiteDeclaration: class,
+		CallSiteMethod:      "Foo",
+		CallOutgoing:        server.RandomLocalWithPulse(),
+		Arguments:           insolar.MustSerialize([]interface{}{}),
+	}
+
+	beforeCount := server.PublisherMock.GetCount()
+	server.SendPayload(ctx, &pl)
+	if !server.PublisherMock.WaitCount(beforeCount+1, 10*time.Second) {
+		t.Fatal("failed to wait until all messages returned")
+	}
+
+	{
+		testutils.WaitSignalsTimed(t, 20*time.Second, executeDone)
+		testutils.WaitSignalsTimed(t, 20*time.Second, server.Journal.WaitAllAsyncCallsDone())
+	}
```

Comment:
```suggestion
	{
		testutils.WaitSignalsTimed(t, 20*time.Second, executeDone)
		testutils.WaitSignalsTimed(t, 20*time.Second, server.Journal.WaitAllAsyncCallsDone())
	}
```

Just two WaitSignals

Reply:
fixed